v1/operator.go

```diff
@@ -0,0 +1,135 @@
+)
+
+type Operator struct {
+	client *Client
+}
+
+func (c *Client) Operator() *Operator {
+	return &Operator{client: c}
+}
+
+func (o *Operator) OperatorApi() *client.OperatorApiService {
+	return o.client.apiClient.OperatorApi
+}
+
+func (o *Operator) Raft(ctx context.Context) (*client.RaftConfiguration, error) {
+	request := o.OperatorApi().GetOperatorRaftConfiguration(o.client.Ctx)
+
+	result, err := o.client.ExecRequest(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+
+	final := result.(client.RaftConfiguration)
+	return &final, nil
+}
+
+func (o *Operator) Peer(ctx context.Context) error {
+	request := o.OperatorApi().DeleteOperatorRaftPeer(o.client.Ctx)
+
+	err := o.client.ExecNoResponseRequest(ctx, request)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *Operator) Autopilot(ctx context.Context) (*client.AutopilotConfiguration, error) {
+	request := o.OperatorApi().GetOperatorAutopilotConfiguration(o.client.Ctx)
+
+	result, err := o.client.ExecRequest(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+
+	final := result.(client.AutopilotConfiguration)
+	return &final, nil
+}
+
+func (o *Operator) UpdateAutopilot(
+	ctx context.Context,
+	cleanupdeadservers bool,
+	lastcontactthreshold string,
+	maxtrailinglogs int32,
+	serverstabilizationtime string,
+	enableredundancyzones bool,
+	disableupgrademigration bool,
+	enablecustomupgrades bool,
+) (*bool, error) {
+	request := o.OperatorApi().PutOperatorAutopilotConfiguration(o.client.Ctx)
+
+	updateRequest := client.NewAutopilotConfiguration()
+	updateRequest.SetCleanupDeadServers(cleanupdeadservers)
+	updateRequest.SetLastContactThreshold(lastcontactthreshold)
+	updateRequest.SetMaxTrailingLogs(maxtrailinglogs)
+	updateRequest.SetServerStabilizationTime(serverstabilizationtime)
+	updateRequest.SetEnableRedundancyZones(enableredundancyzones)
+	updateRequest.SetDisableUpgradeMigration(disableupgrademigration)
+	updateRequest.SetEnableCustomUpgrades(enablecustomupgrades)
+
+	request = request.AutopilotConfiguration(*updateRequest)
+
+	result, err := o.client.ExecRequest(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+
+	final := result.(bool)
+	return &final, nil
+}
+
+func (o *Operator) AutopilotHealth(ctx context.Context) (*client.OperatorHealthReply, error) {
+	request := o.OperatorApi().GetOperatorAutopilotHealth(o.client.Ctx)
+
+	result, err := o.client.ExecRequest(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+
+	final := result.(client.OperatorHealthReply)
+	return &final, nil
+}
+
+func (o *Operator) Scheduler(ctx context.Context) (*client.SchedulerConfigurationResponse, *QueryMeta, error) {
+	request := o.OperatorApi().GetOperatorSchedulerConfiguration(o.client.Ctx)
+
+	result, meta, err := o.client.ExecQuery(ctx, request)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	final := result.(client.SchedulerConfigurationResponse)
+	return &final, meta, nil
+}
+
+// NOTE: there's no RejectJobRegistration
+func (o *Operator) UpdateScheduler(
```

Comment:
I have the same question here.  Is it possible to accept a `SchedulerConfiguration` struct instead of a long list of parameters?

Reply:
and done!